src/TMap.cpp

```diff
@@ -2649,6 +2653,542 @@ void TMap::setRoomNamesShown(bool shown)
+{
+    QString destination{dest};
+    if (!dest.endsWith(QLatin1String(".json"), Qt::CaseInsensitive)) {
+        destination.append(QLatin1String(".json"));
+    }
+
+    if (mpProgressDialog) {
+        return {false, QStringLiteral("import or export already in progress")};
+    }
+
+    mProgressDialogRoomsTotal = mpRoomDB->getRoomMap().count();
+    mProgressDialogAreasTotal = mpRoomDB->getAreaMap().count();
+    QMapIterator<int, TArea*> itArea(mpRoomDB->getAreaMap());
+    mProgressDialogLabelsTotal = 0;
+    while (itArea.hasNext()) {
+        itArea.next();
+        auto pA = itArea.value();
+        if (pA) {
+            mProgressDialogLabelsTotal += pA->mMapLabels.size();
+        }
+    }
+
+    mpProgressDialog = new QProgressDialog(tr("Exporting JSON map data from %1\n"
+                                              "Areas: %2 of: %3   Rooms: %4 of: %5   Labels: %6 of: %7...")
+                                           .arg(mProfileName,
+                                                QLatin1String("0"),
+                                                QString::number(mProgressDialogAreasTotal),
+                                                QLatin1String("0"),
+                                                QString::number(mProgressDialogRoomsTotal),
+                                                QLatin1String("0"),
+                                                QString::number(mProgressDialogLabelsTotal)),
+                                           tr("Abort"),
+                                           0,
+                                           mProgressDialogRoomsTotal,
+                                           mpHost->mpConsole);
+    mpProgressDialog->setValue(0);
+    mpProgressDialog->setWindowModality(Qt::NonModal);
+    mpProgressDialog->setWindowTitle(tr("Map JSON export", "This is a title of a progress window."));
+    mpProgressDialog->setWindowIcon(QIcon(QStringLiteral(":/icons/mudlet_map_download.png")));
+    mpProgressDialog->setMinimumWidth(500);
+    mpProgressDialog->setAutoClose(false);
+    mpProgressDialog->setAutoReset(false);
+    mpProgressDialog->setMinimumDuration(0); // Normally waits for 4 seconds before showing
+    qApp->processEvents();
+    QFile file(destination);
+    if (!file.open(QFile::OpenMode(QFile::Text|QFile::WriteOnly))) {
+        qWarning().noquote().nospace() << "TMap::writeJsonMapFile(...) WARNING - Could not open save file \"" << destination << "\".";
+        mpProgressDialog->setAttribute(Qt::WA_DeleteOnClose, true);
+        mpProgressDialog->close();
+        mpProgressDialog = nullptr;
+        return {false, QStringLiteral("could not open save file \"%1\"").arg(destination)};
+    }
+
+    QJsonObject mapObj;
+    // Use this to track any changes. in a major.minor number format, minor
+    // is to be three digits long.
+    // 0.002 was the first published draft
+    // 0.003 changed the format to encapsulte the room symbol text, it also
```

Comment:
This is a tough sentence to read

Reply:
Trying to keep track of what I changed - and might have broken in each iteration. I'll see if I can be more readable.

Reply:
I've tarted this up a bit and moved it to the start of the method.